src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -1844,6 +1849,27 @@ public bool IsBetweenTypeMembers(SourceText sourceText, SyntaxNode root, int pos
             return false;
         }
 
+        public bool IsOnNamespaceDeclaration(SyntaxNode root, int position)
+        {
+            var token = root.FindToken(position);
+            if (token.Kind() == SyntaxKind.EndOfFileToken)
+            {
+                token = token.GetPreviousToken();
+            }
+
+            var namespaceDecl = token.GetAncestor<NamespaceDeclarationSyntax>();
+            if (namespaceDecl == null)
+            {
+                return false;
+            }
+
+            var start = namespaceDecl.SpanStart;
+            var end = namespaceDecl.Name?.GetLastToken().FullSpan.End ??
+                        namespaceDecl.NamespaceKeyword.FullSpan.End;
+
+            return position >= start && position <= end;
```

Comment:
this is not a syntax fact.

Reply:
A more appropriate syntaxfact would be GetNamespaceDeclarationName.  It would return the right thing for a VB or C# NamespaceDeclarationSyntax, and you coudl then check if your position was in the span of that name.